Added random_mac configuration option to ethernet.

bluecmd opened the pull request:
```
* peripheral/eth.c (eth_random_mac): Added configuration option to
control the MAC address generation from the simulator.
* doc/or1ksim.texi: Added information on random_mac
* configuration.
* doc/or1ksim.info: Regenerated.
* doc/version.info: Regenerated.
* sim.cfg: Updated sample configuration to list random_mac.
```

jeremybennett commented:
Nice work - thanks for this. If possible, please add some appropriate tests to the testsuite (stops someone breaking it in the future). 

Assuming  regression testing and "make distcheck" both pass, please commit and also cherry-pick into the or1k branch.

bluecmd commented:
Will do, thanks.

skristiansson commented:
I've tested to run 'make distcheck' against this and there were no new failures.

bluecmd committed:
Add simple loopback support for ethernet

This patch adds a simple loopback support for the ethernet MAC. It works
by saving the latest packet from TX FD which will then be used to fill
the RX BD. Tested to work for 1 TX and 1 RX BD.
